src/modules/users/users.controller.ts

```diff
@@ -36,7 +38,11 @@ export class UsersController {
   }
 
   @Get()
-  findAll() {
+  findAll(@Query('type') type?: string) {
+    const typeUser = type?.charAt(0).toUpperCase() + type?.slice(1);
+    if (Object.keys(Role).some((v) => v === typeUser)) {
+      return this.usersService.findAll(type);
+    }
     return this.usersService.findAll();
```

Comment:
Maybe I should create a DTO for this and inside that DTO, I should specify a field called Type and this should be of type `UserType`, this should be a enum, that could be MANAGER || TEACHER || STUDENT.